tests/winstanley/ReferenceSpec.scala

```diff
@@ -0,0 +1,25 @@
+package winstanley
+
+import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase
+import winstanley.psi.{WdlDeclaration, WdlTaskDeclaration}
+import winstanley.psi.impl.{WdlDeclarationImpl, WdlTaskDeclarationImpl}
+
+import scala.reflect.ClassTag
+
+class ReferenceSpec extends LightCodeInsightFixtureTestCase {
+
+  def testPositiveTaskReference(): Unit = referenceTest("positive_task_reference.wdl", "hello_task", classOf[WdlTaskDeclarationImpl])
+
+  def testPositiveVariableReference(): Unit = referenceTest("positive_variable_reference.wdl", "i", classOf[WdlDeclarationImpl])
+
+  private def referenceTest(path: String, expectedReferenceText: String, expectedReferenceType: Any): Unit = {
```

Comment:
You can avoid leaving `expectedReferenceType` here untyped with a type argument, eg
```
  def testPositiveVariableReference(): Unit = referenceTest[WdlDeclarationImpl]("positive_variable_reference.wdl", "i")

  private def referenceTest[ExpectedReferenceType](path: String, expectedReferenceText: String): Unit = { ... }
```